ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/DecimalTransformIndicator.java

```diff
@@ -37,140 +37,165 @@
-                break;
-            }
-        }
-
-        else if (simpleType != null) {
-            switch (simpleType) {
-            case sqrt:
-                return val.sqrt();
-            case abs:
-                return val.abs();
-            case log:
-                return numOf(Math.log(val.doubleValue()));
-            default:
-                break;
-            }
-        }
-
-        return val;
-    }
-
-    @Override
-    public String toString() {
-        if (type != null) {
-            return getClass().getSimpleName() + " Coefficient: " + coefficient + " Transform(" + type.name() + ")";
-        }
-        return getClass().getSimpleName() + "Transform(" + simpleType.name() + ")";
-    }
+	private static final long serialVersionUID = -8017034587193428498L;
+
+	/**
+	 * Select the type for transformation.
+	 */
+	public enum DecimalTransformType {
+
+		/**
+		 * Transforms the input indicator by indicator.plus(coefficient).
+		 */
+		plus {
+			@Override
+			Num calculate(Num val, Num coefficient) {
+				return val.plus(coefficient);
+			}
+		},
+
+		/**
+		 * Transforms the input indicator by indicator.minus(coefficient).
+		 */
+		minus {
+			@Override
+			Num calculate(Num val, Num coefficient) {
+				return val.minus(coefficient);
+			}
+		},
+
+		/**
+		 * Transforms the input indicator by indicator.multipliedBy(coefficient).
+		 */
+		multiply {
+			@Override
+			Num calculate(Num val, Num coefficient) {
+				return val.multipliedBy(coefficient);
+			}
+		},
+
+		/**
+		 * Transforms the input indicator by indicator.dividedBy(coefficient).
+		 */
+		divide {
+			@Override
+			Num calculate(Num val, Num coefficient) {
+				return val.dividedBy(coefficient);
+			}
+		},
+
+		/**
+		 * Transforms the input indicator by indicator.max(coefficient).
+		 */
+		max {
+			@Override
+			Num calculate(Num val, Num coefficient) {
+				return val.max(coefficient);
+			}
+		},
+
+		/**
+		 * Transforms the input indicator by indicator.min(coefficient).
+		 */
+		min {
+			@Override
+			Num calculate(Num val, Num coefficient) {
+				return val.min(coefficient);
+			}
+		};
+
+		abstract Num calculate(Num val, Num coefficient);
+	}
+
+	/**
+	 * Select the type for transformation.
+	 */
+	public enum DecimalTransformSimpleType {
+		/**
+		 * Transforms the input indicator by indicator.abs().
+		 */
+		abs {
+			@Override
+			Num calculate(Num val) {
+				return val.abs();
+			}
+		},
+
+		/**
+		 * Transforms the input indicator by indicator.sqrt().
+		 */
+		sqrt {
+			@Override
+			Num calculate(Num val) {
+				return val.sqrt();
+			}
+		},
+
+		/**
+		 * Transforms the input indicator by indicator.log().
+		 */
+		log {
+			@Override
+			Num calculate(Num val) {
+				return val.numOf(Math.log(val.doubleValue()));
+			}
+		};
+
+		abstract Num calculate(Num val);
+	}
+
+	private Indicator<Num> indicator;
+	private Num coefficient;
+	private DecimalTransformType type;
+	private DecimalTransformSimpleType simpleType;
```

Comment:
How can we else avoid this null check?

Reply:
We could split this up into two indicators
- SimpleDecimalTransformIndicator
- DecimalTransformIndicator

Another solution (but imo dirty) could be to merge `DecimalTransformType ` and `DecimalTransformSimpleType` into one enum, and the functions of `DecimalTransformSimpleType` are ignoring the coefficient. And coefficient could get a default value = 0

Btw. MultiplierIndicator.java should also be removed by this indicator, or?

Reply:
Splitting into two classes only to avoid the null-check? I had think about it in the past when I created this indicator and came to the conclusion that it is better to have one indicator dealing with "transforming values by math operations". 

And yes, you can remove `MultiplierIndicator.java` because it is already handled by this indicator.

Reply:
>Another solution (but imo dirty) could be to merge DecimalTransformType and DecimalTransformSimpleType into one enum, and the functions of DecimalTransformSimpleType are ignoring the coefficient. And coefficient could get a default value = 0

I can do this. I think this solution would be better than splitting the indicator into two classes. Should I do this?

Reply:
>Another solution (but imo dirty) could be to merge DecimalTransformType and DecimalTransformSimpleType into one enum, and the functions of DecimalTransformSimpleType are ignoring the coefficient. And coefficient could get a default value = 0

Should I do this?

Reply:
> We could split this up into two indicators

I can do this by another PR.